internal/cmd/agent.go

```diff
@@ -0,0 +1,137 @@
+package cmd
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"io/fs"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"sync"
+	"time"
+
+	"github.com/shirou/gopsutil/v3/process"
+	"github.com/spf13/cobra"
+
+	"github.com/infrahq/infra/internal/logging"
+	"github.com/infrahq/infra/internal/repeat"
+)
+
+func newAgentCmd() *cobra.Command {
+	return &cobra.Command{
+		Use:    "agent",
+		Short:  "Start the Infra agent",
+		Long:   "Start the Infra agent that runs to sync Infra state",
+		Args:   NoArgs,
+		Hidden: true,
+		RunE: func(cmd *cobra.Command, _ []string) error {
+			var wg sync.WaitGroup
+
+			// start background tasks
+			ctx, cancel := context.WithCancel(context.Background())
+			defer cancel()
+
+			repeat.InGroup(&wg, ctx, cancel, 1*time.Minute, syncKubeConfig)
+			// add the next agent task here
+
+			wg.Wait()
+
+			return ctx.Err()
+		},
+	}
+}
+
+// configAgentRunning checks if the agent process stored in config is still running
+func configAgentRunning() (bool, error) {
+	pid, err := readStoredAgentProcessID()
+	if err != nil {
+		if errors.Is(err, fs.ErrNotExist) {
+			// this is the first time the agent is running, suppress the error and continue
+			logging.S.Debug(err)
+			return false, nil
+		}
+		return false, err
+	}
+
+	return processRunning(int32(pid))
+}
+
+func processRunning(pid int32) (bool, error) {
+	if pid == 0 { // on windows pid 0 is the system idle process, it will be running but its not the agent
+		return false, nil
+	}
+
+	running, err := process.PidExists(pid)
+	if err != nil {
+		return false, err
+	}
+
+	return running, nil
+}
+
+// execAgent executes the agent in the background and stores its process ID in configuration
+func execAgent() error {
+	cmd := exec.Command("infra", "agent")
```

Comment:
This will fail if `infra` is not in `PATH` or if the `infra` in `PATH` is actually a different binary